scripts/globals/events/egg_hunt_egg-stravaganza.lua

```diff
@@ -0,0 +1,898 @@
+        },
+    },
+
+    [xi.zone.WINDURST_WOODS] =
+    {
+        cs = 785,
+        decorations =
+        {
+            { 128,  107.630, -5.000, -33.200, "0x00006D0500000000000000000000000000000000", },
+            { 128,  107.630, -5.000, -46.800, "0x00006D0500000000000000000000000000000000", },
+            {   0, -107.000, -5.250,  35.500, "0x00006D0500000000000000000000000000000000", },
+            {   0, -107.000, -5.250,  44.500, "0x00006D0500000000000000000000000000000000", },
+        },
+        helpers =
+        {
+            {   0, -105.500, -5.250,  35.500, "0x01001E0282117020173017401750006000700000", },
+            {   0, -105.500, -5.250,  44.500, "0x01001E0782117020173017401750006000700000", },
+            { 127,  106.500, -5.000, -33.200, "0x01001E0382117020173017401750006000700000", },
+            { 127,  106.500, -5.000, -46.800, "0x01001E0482117020173017401750006000700000", },
+            { 127, -155.000, -4.000, -27.000, "0x01001E0100107120173017401750006000700000", },
+            { 127, -155.000, -4.000, -33.000, "0x01001E0282117120173017401750006000700000", },
+        },
+    },
+}
+
+local messageOffset =
+{
+    REWARD1 = 0,  -- Egg-cellent! Here's your prize, kupo! Now if only somebody would bring me a super combo... Oh, egg-scuse me! Forget I said that, kupo!
+    REWARD2 = 1,  -- Egg-straordinary! Here's your special prize, kupo! But how did you figure out the secret combination...?
+    REWARD3 = 2,  -- How in-eggs-plicably astounding! You're truly a hard-boiled detective of the highest degree, kupo! Here's your prize, and don't forget to enjoy the rest of the festivities.
+    WAITING = 3,  -- Hey, don't get all egg-cited! You can only claim one prize a day, kupo!
+    HINTING = 4,  -- These initial eggs are really egg-spensive, kupo! If you want one I'm going to need a little more incentive. No egg-ceptions!
+    TRADE1  = 5,  -- Eggs-actly what I wanted! Here's your reward, kupo!
+    TRADE2  = 6,  -- Ku-kupo! Such an egg-stravagant item, for little old me? Here's a whole basketful of initial eggs for you, kupo!
+    TRADE3  = 7,  -- My eyes! My eyes! Get out of here, and take those egg-ceptionally nasty things with you, kupo!
+    TRADE4  = 8,  -- Kupo... Kupo! Please! No more, kupo! Here, take these and get out of here, kupo!
+    DELAY   = 9,  -- Hey, I can only trade with you once a day. We wouldn't want to make the egg hunt over easy, now, kupo!
+    HELP    = 10, -- Hey, you look new here, kupo! If you need anything, just let me know!
+    REWARD4 = 11, -- Heh heh heh... I have something egg-specially egg-citing in store for you, kupo!
+}
+
+------------------------------------
+-- Helpers
+------------------------------------
+
+xi.events.egg_hunt.charToEgg = function(char)
+    local letter = string.byte(string.lower(char)) - 97
+    return xi.items.A_EGG + letter
+end
+
+xi.events.egg_hunt.eggInList = function(eggs, egg)
```

Comment:
I would prefer some more descriptive variable names for eggs, egg, and j in this case, even though j is an iterator.  The function is small enough to understand, but just thinking about future folks